app/models/manageiq/providers/redfish/inventory/parser/physical_infra_manager.rb

```diff
@@ -99,5 +114,51 @@ def get_storage_sum(server)
     def get_storage_capacity(storage)
       storage.Drives.reduce(0) { |acc, d| acc + (d.CapacityBytes || 0) }
     end
+
+    def physical_racks
+      collector.physical_racks.each do |r|
+        persister.physical_racks.build(
+          :ems_ref => r["@odata.id"],
+          :name    => r.Id
+        )
+      end
+    end
+
+    def physical_chassis
+      collector.physical_chassis.each do |c|
+        chassis = persister.physical_chassis.build(
+          :ems_ref      => c["@odata.id"],
+          :health_state => c.Status.Health,
+          :name         => c.Id,
+          :type         => "ManageIQ::Providers::Redfish::PhysicalInfraManager::PhysicalChassis",
+        )
+
+        parent_id = c.dig("Links", "ContainedBy", "@odata.id")
+        next unless parent_id
+
+        parent_chassis = persister.physical_chassis.lazy_find(parent_id)
+        chassis.assign_attributes(:parent_physical_chassis => parent_chassis)
```

Comment:
Similar comment here you should be able to just pass these into the build, instead of the next you can just do `parent_chassis = persister.physical_chassis.lazy_find(parent_id) if parent_id`